dev-qt/qt6-base/patches/qt6_base-6.9.2.patchset

```diff
@@ -0,0 +1,1032 @@
+     // Disconnect from the controlling TTY. This probably won't fail. Must be
+     // done after the session settings from above.
+     if (params.flags.testFlag(QProcess::UnixProcessFlag::DisconnectControllingTerminal)) {
+@@ -883,6 +884,7 @@ static const char *applyProcessParameters(const QProcess::UnixProcessParameters
+             }
+         }
+     }
++#endif
+ 
+     // Disable core dumps near the end. This isn't expected to fail.
+     if (params.flags.testFlag(QProcess::UnixProcessFlag::DisableCoreDumps)) {
+-- 
+2.48.1
+
+
+From 547b99e7842d9f1913ab0d4e786be30d2918841d Mon Sep 17 00:00:00 2001
+From: Gerasim Troeglazov <3dEyes@gmail.com>
+Date: Fri, 4 Oct 2024 18:22:17 +1000
+Subject: Use ICU backend for QTimeZone
+
+
+diff --git a/src/corelib/time/qtimezone.cpp b/src/corelib/time/qtimezone.cpp
+index 906a42f..d4fdc49 100644
+--- a/src/corelib/time/qtimezone.cpp
++++ b/src/corelib/time/qtimezone.cpp
+@@ -26,6 +26,8 @@ static QTimeZonePrivate *newBackendTimeZone()
+     return new QChronoTimeZonePrivate();
+ #elif defined(Q_OS_DARWIN)
+     return new QMacTimeZonePrivate();
++#elif defined(Q_OS_HAIKU)
++    return new QUtcTimeZonePrivate();
+ #elif defined(Q_OS_ANDROID)
+     return new QAndroidTimeZonePrivate();
+ #elif defined(Q_OS_UNIX)
+-- 
+2.48.1
+
+
+From 66100dafa96f8bf920eecd4a42cc8d17809f03ed Mon Sep 17 00:00:00 2001
+From: Gerasim Troeglazov <3dEyes@gmail.com>
+Date: Sat, 5 Oct 2024 19:48:13 +1000
+Subject: Use ICU backend for ianaId
+
+
+diff --git a/src/corelib/time/qtimezone.cpp b/src/corelib/time/qtimezone.cpp
+index d4fdc49..c059f47 100644
+--- a/src/corelib/time/qtimezone.cpp
++++ b/src/corelib/time/qtimezone.cpp
+@@ -49,6 +49,8 @@ static QTimeZonePrivate *newBackendTimeZone(const QByteArray &ianaId)
+     return new QChronoTimeZonePrivate(ianaId);
+ #elif defined(Q_OS_DARWIN)
+     return new QMacTimeZonePrivate(ianaId);
++#elif defined(Q_OS_HAIKU)
++    return new QUtcTimeZonePrivate(ianaId);
+ #elif defined(Q_OS_ANDROID)
+     return new QAndroidTimeZonePrivate(ianaId);
+ #elif defined(Q_OS_UNIX)
+-- 
+2.48.1
+
+
+From 221e0f18dd59c97a3b4dcb1d3f51b5763349a477 Mon Sep 17 00:00:00 2001
+From: Gerasim Troeglazov <3dEyes@gmail.com>
+Date: Sat, 5 Oct 2024 19:48:35 +1000
+Subject: Disable tz backend for Haiku
+
+
+diff --git a/src/corelib/CMakeLists.txt b/src/corelib/CMakeLists.txt
+index 5411bb5..51936d2 100644
+--- a/src/corelib/CMakeLists.txt
++++ b/src/corelib/CMakeLists.txt
+@@ -964,7 +964,7 @@ qt_internal_extend_target(Core
+ qt_internal_extend_target(Core
+     CONDITION
+         QT_FEATURE_timezone AND UNIX
+-        AND NOT ANDROID AND NOT APPLE AND NOT QT_FEATURE_timezone_tzdb
++        AND NOT ANDROID AND NOT APPLE AND NOT HAIKU AND NOT QT_FEATURE_timezone_tzdb
+     SOURCES
+         time/qtimezoneprivate_tz.cpp
+ )
+-- 
+2.48.1
+
+From 6f0d27d2e4ba5fa6562f738aaaf8eaf98ebf51e7 Tue Aug 26 00:00:00 2025
+From: Ken Mays <kmays2000@gmail.com>
+Date: Wed, 3 Sept 2025 05:35:11 +0600
+Subject: Haiku CLOEXEC
```

Comment:
Is this required? there is a `#ifdef CLOSE_RANGE_CLOEXEC` for this, if this isn't defined on Haiku it would be obsolete, or am I wrong here?

Reply:
close_range defined in the BSD unistd.h for -D_DEFAULT_SOURCE.
See: @korli

Reply:
OK, this has been added after beta5, so this should only apply on nightly or when beta6 is out?

Also, wouldn't this be empty if the ifdef isn't sattisfied?
```
#ifdef CLOSE_RANGE_CLOEXEC
        // Mark the file descriptors for closing upon execve() - we delay
        // closing so we don't close the ones QProcess needs for itself.
        // On FreeBSD, this probably won't fail.
        // On Linux, this will fail with ENOSYS before kernel 5.9 and EINVAL
        // before 5.11.
        r = close_range(fd, INT_MAX, CLOSE_RANGE_CLOEXEC);
#endif
```